app/controllers/application_controller.rb

```diff
@@ -3,4 +3,16 @@ class ApplicationController < ActionController::Base
   # For APIs, you may want to use :null_session instead.
   protect_from_forgery with: :exception
 
+  def setup_cart
+    @cart = current_user.current_cart
+
+    if @cart.nil?
+      @cart = Cart.create
```

Comment:
Prefer to make the cart already associated with the current user via `current_user.carts.build`.